components/MapContainer/MapContainer.tsx

```diff
@@ -0,0 +1,231 @@
+import React from 'react';
+import { Select } from '../Select';
+import { Map } from '../Map';
+import ugandaDistricts from './geoJSON/district.json';
+import ugandasubcounties from './geoJSON/subcounty.json';
+import * as distance from 'jaro-winkler';
+import * as turf from '@turf/turf';
+
+const mapContainerStyle = {
+  padding: '50px'
+};
+
+interface State {
+  leaflet: any;
+  map: any;
+  selectedDistrict: string;
+  boundaryType: string;
+  subcountyDropdownOptions: any[];
+  selectedSubcounty: string;
+}
+
+class MapContainer extends React.Component<{}, State> {
+  constructor(props: any) {
+    super(props);
+    this.state = {
+      leaflet: {},
+      map: {},
+      selectedDistrict: '',
+      selectedSubcounty: '',
+      boundaryType: 'all',
+      subcountyDropdownOptions: []
+    };
+    this.handleSubcountyChange = this.handleSubcountyChange.bind(this);
+    this.handleDistrictChange = this.handleDistrictChange.bind(this);
+    this.initialiseMapState = this.initialiseMapState.bind(this);
+  }
+
+  initialiseMapState(L: any, map: any) {
+    this.setState({ leaflet: L, map }, () => {
+      this.addLayer();
+    });
+  }
+
+  loadDistrictSelect(districts: any) {
+    const options = [];
+    for (const district in districts.features) {
+      if (districts.features) {
+        options.push({
+          value: districts.features[district].properties.DNAME2014,
+          label: districts.features[district].properties.DNAME2014
+        });
+      }
+    }
+
+    return options;
+  }
+
+  loadSubcountySelect(subcounties: any) {
+    const options = [];
+    for (const subcounty in subcounties) {
+      if (subcounties[subcounty]) {
+        options.push({
+          value: subcounties[subcounty].properties.SName2016,
+          label: subcounties[subcounty].properties.SName2016
+        });
+      }
+    }
+
+    return options;
+  }
+
+  handleDistrictChange(selectedOption: any) {
+    const districtSubcounties = this.findSelectedDistrictSubcounties(selectedOption.value, ugandasubcounties);
+    const subcountyOptions = this.loadSubcountySelect(districtSubcounties);
+
+    this.setState({
+      selectedDistrict : selectedOption.value,
+      boundaryType : 'district',
+      subcountyDropdownOptions: subcountyOptions
+    }, () => {
+      this.addLayer();
+    });
+  }
+
+  handleSubcountyChange(selectedOption: any) {
+    this.setState({
+      selectedSubcounty : selectedOption.value,
+      boundaryType : 'subcounty'
+    }, () => {
+      this.addLayer();
+    });
+  }
+
+  addLayer() {
+    switch (this.state.boundaryType) {
```

Comment:
Switch statements are anti-pattern ... below is a good resource on the common js code "smells" https://elijahmanor.com/javascript-smells/

Reply:
Updated to use `if` and `else` statements